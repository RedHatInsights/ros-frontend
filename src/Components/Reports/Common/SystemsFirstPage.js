import React, { Fragment } from 'react';
import propTypes from 'prop-types';
import { Section, Column } from '@redhat-cloud-services/frontend-components-pdf-generator';
import { SystemsTablePage } from './SystemsTablePage';

export const SystemsFirstPage = ({ data, totalSystems, filterText }) => {

    return <Fragment key="first-section">
        <Section>
            <Column>
                {`This report identified ${totalSystems} ${totalSystems > 1 ? 'RHEL systems' : 'RHEL system' }. ${filterText}`}
            </Column>
        </Section>
        <SystemsTablePage data={data} page={0} />
    </Fragment>;
};

SystemsFirstPage.propTypes = {
    data: propTypes.array,
    totalSystems: propTypes.number,
    filterText: propTypes.string
};
