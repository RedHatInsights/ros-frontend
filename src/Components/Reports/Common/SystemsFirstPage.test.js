import React from 'react';
import renderer from 'react-test-renderer';
import { SystemsFirstPage } from './SystemsFirstPage';

describe('Systems First Page Component', () => {

    it('should matches snapshot for systems first page and renders correctly', () => {

        const firstPageProps = {
            data: [
                ['iqe-ros-bb519f81', 'N/A', 'RHEL 8.4', '90%', '80%', '0.314', '2', 'Undersized'],
                ['iqe-ros-bb519f81', 'team-ros', 'RHEL 8.4', '90%', '80%', '0.314', '2', 'Under pressure'],
                ['iqe-ros-bb519f81', 'N/A', 'RHEL 8.4', 'N/A', 'N/A', 'N/A', '2', 'Waiting for data']
            ],
            totalSystems: 3,
            filterText: `Filters applied\nState: Undersized,Under pressure,Waiting for data\nName: iqe`
        };

        const component = renderer.create(<SystemsFirstPage {...firstPageProps} />).toJSON();

        expect(component).toMatchSnapshot();

    });

    it('should matches snapshot for systems first page and renders correctly with 1 system', () => {

        const firstPageProps = {
            data: [
                ['iqe-ros-bb519f81', 'team-ros', 'RHEL 8.4', '90%', '80%', '0.314', '2', 'Undersized']
            ],
            totalSystems: 1,
            filterText: `Filters applied\nState: Undersized\nName: iqe\nGroups:team-ros`
        };

        const component = renderer.create(<SystemsFirstPage {...firstPageProps} />).toJSON();

        expect(component).toMatchSnapshot();

    });

});
