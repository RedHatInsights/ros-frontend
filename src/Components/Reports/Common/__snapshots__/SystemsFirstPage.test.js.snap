// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Systems First Page Component should matches snapshot for systems first page and renders correctly 1`] = `
Array [
  <VIEW
    style={
      Object {
        "margin": "7 0",
      }
    }
  >
    <TEXT
      style={
        Object {
          "color": "#c9190b",
          "fontSize": 9,
          "fontWeight": 700,
          "margin": "7 0",
        }
      }
    >
      
    </TEXT>
    <VIEW
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <VIEW
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <TEXT>
          This report identified 3 RHEL systems. Filters applied
State: Undersized,Under pressure,Waiting for data
Name: iqe
        </TEXT>
      </VIEW>
    </VIEW>
  </VIEW>,
  <VIEW
    style={
      Object {
        "margin": "7 0",
      }
    }
  >
    <TEXT
      style={
        Object {
          "color": "#c9190b",
          "fontSize": 9,
          "fontWeight": 700,
          "margin": "7 0",
        }
      }
    >
      
    </TEXT>
    <VIEW
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <VIEW
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <VIEW
          style={Object {}}
        >
          <VIEW
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": "0.5rem",
                "paddingLeft": "0.5rem",
                "paddingTop": "0.5rem",
              }
            }
          >
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 2.5,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 4,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Name
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Group
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "textAlign": "left",
                  },
                ]
              }
            >
              OS
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "textAlign": "left",
                  },
                ]
              }
            >
              CPU utilization
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Memory utilization
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "textAlign": "left",
                  },
                ]
              }
            >
              I/O utilization
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Suggestions
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "textAlign": "left",
                  },
                ]
              }
            >
              State
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1.2,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingLeft": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Last reported
            </TEXT>
          </VIEW>
          <VIEW
            style={
              Object {
                "borderBottomColor": "#8a8d90",
                "borderBottomStyle": "solid",
                "borderBottomWidth": 1,
                "borderTopColor": "#8a8d90",
                "borderTopStyle": "solid",
                "borderTopWidth": 1,
              }
            }
          >
            <VIEW
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": "0.5rem",
                  "paddingLeft": "0.5rem",
                  "paddingTop": "0.5rem",
                }
              }
            >
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 2.5,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingRight": 4,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                iqe-ros-bb519f81
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                N/A
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                RHEL 8.4
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                90%
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                80%
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                0.314
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                2
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Undersized
              </TEXT>
            </VIEW>
            <VIEW
              style={
                Object {
                  "backgroundColor": "#f0f0f0",
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": "0.5rem",
                  "paddingLeft": "0.5rem",
                  "paddingTop": "0.5rem",
                }
              }
            >
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 2.5,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingRight": 4,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                iqe-ros-bb519f81
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                team-ros
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                RHEL 8.4
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                90%
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                80%
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                0.314
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                2
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Under pressure
              </TEXT>
            </VIEW>
            <VIEW
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": "0.5rem",
                  "paddingLeft": "0.5rem",
                  "paddingTop": "0.5rem",
                }
              }
            >
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 2.5,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingRight": 4,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                iqe-ros-bb519f81
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                N/A
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                RHEL 8.4
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                N/A
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                N/A
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                N/A
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                2
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Waiting for data
              </TEXT>
            </VIEW>
          </VIEW>
        </VIEW>
      </VIEW>
    </VIEW>
  </VIEW>,
]
`;

exports[`Systems First Page Component should matches snapshot for systems first page and renders correctly with 1 system 1`] = `
Array [
  <VIEW
    style={
      Object {
        "margin": "7 0",
      }
    }
  >
    <TEXT
      style={
        Object {
          "color": "#c9190b",
          "fontSize": 9,
          "fontWeight": 700,
          "margin": "7 0",
        }
      }
    >
      
    </TEXT>
    <VIEW
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <VIEW
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <TEXT>
          This report identified 1 RHEL system. Filters applied
State: Undersized
Name: iqe
Groups:team-ros
        </TEXT>
      </VIEW>
    </VIEW>
  </VIEW>,
  <VIEW
    style={
      Object {
        "margin": "7 0",
      }
    }
  >
    <TEXT
      style={
        Object {
          "color": "#c9190b",
          "fontSize": 9,
          "fontWeight": 700,
          "margin": "7 0",
        }
      }
    >
      
    </TEXT>
    <VIEW
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <VIEW
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <VIEW
          style={Object {}}
        >
          <VIEW
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingBottom": "0.5rem",
                "paddingLeft": "0.5rem",
                "paddingTop": "0.5rem",
              }
            }
          >
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 2.5,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingRight": 4,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Name
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Group
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "textAlign": "left",
                  },
                ]
              }
            >
              OS
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "textAlign": "left",
                  },
                ]
              }
            >
              CPU utilization
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Memory utilization
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "textAlign": "left",
                  },
                ]
              }
            >
              I/O utilization
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Suggestions
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1,
                    "fontSize": 8,
                    "textAlign": "left",
                  },
                ]
              }
            >
              State
            </TEXT>
            <TEXT
              style={
                Array [
                  Object {
                    "flex": 1.2,
                    "fontSize": 8,
                    "paddingBottom": 2,
                    "paddingLeft": 2,
                    "paddingTop": 2,
                    "textAlign": "left",
                  },
                ]
              }
            >
              Last reported
            </TEXT>
          </VIEW>
          <VIEW
            style={
              Object {
                "borderBottomColor": "#8a8d90",
                "borderBottomStyle": "solid",
                "borderBottomWidth": 1,
                "borderTopColor": "#8a8d90",
                "borderTopStyle": "solid",
                "borderTopWidth": 1,
              }
            }
          >
            <VIEW
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": "0.5rem",
                  "paddingLeft": "0.5rem",
                  "paddingTop": "0.5rem",
                }
              }
            >
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 2.5,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingRight": 4,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                iqe-ros-bb519f81
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                team-ros
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                RHEL 8.4
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                90%
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                80%
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                0.314
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                2
              </TEXT>
              <TEXT
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "fontSize": 8,
                      "paddingBottom": 2,
                      "paddingLeft": 2,
                      "paddingTop": 2,
                      "textAlign": "left",
                    },
                  ]
                }
              >
                Undersized
              </TEXT>
            </VIEW>
          </VIEW>
        </VIEW>
      </VIEW>
    </VIEW>
  </VIEW>,
]
`;
